.travis.yml

```diff
@@ -1,13 +1,41 @@
 # copyright © 2017 the libfulltext authors (see AUTHORS.md and LICENSE)
 
+sudo: false
 language: python
+cache: pip
 python:
-    - "3.5"
-    - "3.6"
-before_install:
-    - pip install -r requirements.txt
-    - pip install pylint nose2
-script:
-    - pylint bin libfulltext
-    - nose2
-    - pip install ${TRAVIS_BUILD_DIR}
+  - "3.5"
+  - "3.6"
+
+install:
+  - travis_retry pip install -r requirements.txt
+  - travis_retry pip install nose2 pylint flake8
+
+script:  # This is the 'test' build stage
+  - nose2 --with-coverage
+    # Fail if coverage below threshold
+  - python3 -m coverage report --fail-under=55 > /dev/null
+  - pip install ${TRAVIS_BUILD_DIR}
+
+after_success:
+  - echo Success  # TODO upload coverage
+
+# These jobs are only run with the first python version
+# mentioned in the initial list.
+jobs:
+  include:
+    - stage: code style
+      script: pylint bin libfulltext
+      env: CODE_STYLE="pylint"
+    - script: flake8
+      env: CODE_STYLE="flake8"
+    #
+    - stage: deploy
+      # TODO deploy to pip
+      script: skip
+
+stages:
+  - code style
+  - test
+  - name: deploy
+    if: branch = master AND type = push
```

Comment:
Are these jobs+stages required for the coverage?

Reply:
Not necessarily, but it makes the reaction time of the CI faster: First code standards are checked and only if the code complies the more time-consuming unit tests are executed on multiple python versions.